test_common.go

```diff
@@ -131,30 +138,58 @@ func signerFromJWK(t *testing.T, j string) *cose.Signer {
 	ks, err := jwk.ParseString(j)
 	require.Nil(t, err)
 
-	var key crypto.PrivateKey
+	var key crypto.Signer
 
 	err = ks.Keys[0].Raw(&key)
 	require.Nil(t, err)
 
 	var crv elliptic.Curve
-	var alg *cose.Algorithm
+	var alg cose.Algorithm
 
 	switch v := key.(type) {
 	case *ecdsa.PrivateKey:
 		crv = v.Curve
 		if crv == elliptic.P256() {
-			alg = cose.ES256
+			alg = cose.AlgorithmES256
 			break
 		}
 		require.True(t, false, "unknown elliptic curve %v", crv)
 	default:
 		require.True(t, false, "unknown private key type %v", reflect.TypeOf(key))
 	}
 
-	s, err := cose.NewSignerFromKey(alg, key)
+	s, err := cose.NewSigner(alg, key)
 	require.Nil(t, err)
 
-	return s
+	return &s
+}
+
+func pubKeyFromJWK(t *testing.T, j string) crypto.PublicKey {
```

Comment:
It looks like this and `signerFromJWK` share 99% of the logics and should be refactored.

Reply:
Agree and have now refactored! Looks nice and crisp!